src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -119,6 +119,29 @@ export class GridOutputComponent extends AngularDisposable implements IMimeCompo
 		}
 		if (!this._table) {
 			let source = <IDataResource><any>this._bundleOptions.data[this.mimeType];
+			let columnNames: Array<string> = [];
+			let rowIndex = 0;
+			// Get schema list of grid
+			for (let i of source.schema.fields) {
+				columnNames.push(i.name);
+			}
+			// Check to see if data source is ordered properly based on schema
+			// SQL notebooks do not use columnName as key (instead uses indices)
+			// So we add a condition for SQL notebooks to not be reordered
+			if (source.data.length > 0) {
+				if (columnNames !== Object.keys(source.data[0]) && Object.keys(source.data[0])[0] !== '0') {
```

Comment:
Would suggest explaining what the check against '0' is for - it's not clear to me what that value represents or why it would mean you don't need to re-order the columns.

Reply:
This condition is made since SQL notebooks data within the application/vnd.dataresource+json is sorted via indices instead of columnNames, so I am unable to make a comparison with columnNames. I am trying to find a better way to make a condition to exclude SQL notebooks.

Reply:
This is now done through the index format condition that we check to see if the data is formatted in indices and if so then we know the data is already formatted correctly. We only reorder tables that contain string formatted keys, which would never be SQL notebooks.